python/private/python_repositories.bzl

```diff
@@ -191,6 +191,22 @@ def _python_repository_impl(rctx):
     elif rctx.attr.distutils_content:
         rctx.file(distutils_path, rctx.attr.distutils_content)
 
+    if "darwin" in platform:
+        # Fix up the Python distribution's LC_ID_DYLIB field.
+        # It points to a build directory local to the GitHub Actions
+        # host machine used in the Python standalone build, which causes
+        # dyld lookup errors. To fix, set the full path to the dylib as
+        # it appears in the Bazel workspace as its LC_ID_DYLIB using
+        # the `install_name_tool` bundled with macOS.
+        dylib = "lib/libpython{}.dylib".format(python_short_version)
+        full_dylib_path = rctx.path(dylib)
+        repo_utils.execute_checked(
+            rctx,
+            op = "python_repository.FixUpDyldIdPath",
+            arguments = [repo_utils.which_checked(rctx, "install_name_tool"), "-id", full_dylib_path, dylib],
```

Comment:
This fix is only possible if the repository platform and host platform are the same.

Is it possible to solve this in a different way such that we can have correct behaviour when cross-building (e.g. building on a linux host platform with a configured toolchain to build against a mac platform?

Reply:
I'm not sure, since the dylib format is very Apple-specific. For example, according to this thread: https://forums.developer.apple.com/forums/thread/736719, at the very bottom, `install_name_tool` is mentioned as the canonical way of fixing up rpaths (and also IDs) after the build.

Another question is what to do when using another interpreter's lib - maybe the solution is rather setting `@rpath/lib/libpython3.11.dylib`, so that explicitly setting the RPATH to an interpreter (like e.g. `python@3.11` from something like Homebrew) would work.